matchzoo/metrics.py

```diff
@@ -1,190 +1,254 @@
-
-    Example:
-        >>> rs = [[0, 0, 0, 1], [1, 0, 0], [1, 0, 0]]
-        >>> mean_reciprocal_rank(rs)
-        0.75
-
-    :param rs: Iterator of relevance scores (list or numpy) in rank order
-            (first element is the first item).
-    :return: Mean reciprocal rank.
-    """
-    rs = (np.asarray(r).nonzero()[0] for r in rs)
-    return np.mean([1. / (r[0] + 1) if r.size else 0. for r in rs])
-
-
-def r_precision(r: list) -> float:
-    """
-    Calculate precision after all relevant documents have been retrieved.
-
-    Relevance is binary (nonzero is relevant).
-
-    Example:
-        >>> r = [0, 1, 0]
-        >>> r_precision(r)
-        0.5
-        >>> r = [1, 0, 0]
-        >>> r_precision(r)
-        1.0
-        >>> r = [0, 0, 0]
-        >>> r_precision(r)
-        0.0
-
-    :param r: Relevance scores (list or numpy) in rank order
-              (first element is the first item).
-    :return: Precision score.
-    """
-    r = np.asarray(r) != 0
-    z = r.nonzero()[0]
-    if not z.size:
-        return 0.
-    return np.mean(r[:z[-1] + 1])
-
-
-def precision_at_k(r: list, k: int) -> float:
-    """
-    Calculate precision@k.
-
-    Relevance is binary (nonzero is relevant).
-
-    Example:
-        >>> r = [0, 0, 1]
-        >>> precision_at_k(r, 1)
-        0.0
-        >>> precision_at_k(r, 2)
-        0.0
-        >>> precision_at_k(r, 4)
-        Traceback (most recent call last):
-            File "<stdin>", line 1, in ?
-        ValueError: Relevance score length < k
-
-
-    :param r: Relevance scores (list or numpy) in rank order
-              (first element is the first item)
-    :return: Precision @ k
-    :raises: ValueError: len(r) must be >= k.
-    """
-    assert k >= 1
-    r = np.asarray(r)[:k] != 0
-    if r.size != k:
-        raise ValueError('Relevance score length < k')
-    return np.mean(r)
-
-
-def average_precision(r: list):
-    """
-    Calculate average precision (area under PR curve).
-
-    Relevance is binary (nonzero is relevant).
-
-    Example:
-        >>> r = [0, 1, 0, 0]
-        >>> average_precision(r)
-        0.5
-        >>> r = []
-        >>> average_precision(r)
-        0.0
-
-    :param r: Relevance scores (list or numpy) in rank order
-              (first element is the first item).
-    :return: Average precision.
-    """
-    r = np.asarray(r) != 0
-    out = [precision_at_k(r, k + 1) for k in range(r.size) if r[k]]
-    if not out:
+from matchzoo import engine
+
+
+def sort_and_couple(labels: list, scores: np.array) -> list:
+    """Zip the `labels` with `scores` into a single list."""
+    labels = np.squeeze(labels).tolist()
+    scores = np.squeeze(scores).tolist()
+    couple = list(zip(labels, scores))
+    random.shuffle(couple)
+    sorted_couple = sorted(couple, key=lambda x: x[1], reverse=True)
+    return sorted_couple
+
+
+class MeanReciprocalRank(engine.BaseMetric):
+    ALIAS = ['mean_reciprocal_rank', 'mrr', 'MRR']
```

Comment:
shall we create a function in base class to convert alias to lowercase so that we do not necessarily need to write   'mrr'  and 'MRR' (also for other sub-classes).
It's a bit tricky since some times people use `normalized_discounted_cumulative_gain`, sometimes `ndcg`, `NDCG` or even `nDCG`

Reply:
That's right~

Reply:
Fixed